gensim/similarities/levenshtein.py

```diff
@@ -0,0 +1,115 @@
+"""
+
+import logging
+
+# If python-Levenshtein is available, import it.
+# If python-Levenshtein is unavailable, ImportError will be raised in levsim.
+try:
+    import Levenshtein
+    LEVENSHTEIN_EXT = True
+except ImportError:
+    LEVENSHTEIN_EXT = False
+
+from gensim.similarities.termsim import TermSimilarityIndex
+
+logger = logging.getLogger(__name__)
+
+
+def levsim(t1, t2, alpha=1.8, beta=5.0, max_distance=float("inf")):
+    """Get the Levenshtein similarity between two terms.
+
+    Return the Levenshtein similarity between two terms. The similarity is a
+    number between <0.0, 1.0>, higher is more similar.
+
+    Parameters
+    ----------
+    t1 : {bytes, str, unicode}
+        The first compared term.
+    t2 : {bytes, str, unicode}
+        The second compared term.
+    alpha : float
+        The multiplicative factor alpha defined by Charlet and Damnati (2017).
+    beta : float
+        The exponential factor beta defined by Charlet and Damnati (2017).
+    max_distance : {int,float}
+        The maximum Levenshtein distance between t1, and t2. Larger distances
+        will result in the Levenshtein similarity of zero.
+
+    Returns
+    -------
+    float
+        The Levenshtein similarity between `t1` and `t2`.
+
+    Notes
+    -----
+    This notion of Levenshtein similarity was first defined in section 2.2 of
+    `Delphine Charlet and Geraldine Damnati, "SimBow at SemEval-2017 Task 3:
+    Soft-Cosine Semantic Similarity between Questions for Community Question
+    Answering", 2017 <http://www.aclweb.org/anthology/S/S17/S17-2051.pdf>`__.
+
+    """
+    if not LEVENSHTEIN_EXT:
+        raise ImportError("Please install python-Levenshtein Python package to compute the Levenshtein distance.")
+    distance = Levenshtein.distance(t1, t2)
+    if distance >= max_distance:  # This allows future optimization, where Levenshtein.distance terminates early
+        similarity = 0.0
+    else:
+        similarity = alpha * (1 - distance * 1.0 / max(len(t1), len(t2)))**beta
+    return similarity
+
+
+class LevenshteinSimilarityIndex(TermSimilarityIndex):
+    """
+    Computes Levenshtein similarities between terms and retrieves most similar
+    terms for a given term.
+
+    Notes
+    -----
+    This is a naive implementation that iteratively computes pointwise Levenshtein similarities
+    between individual terms. Using this implementation to compute the similarity of all terms in
+    real-world dictionaries such as the English Wikipedia will take years.
+
+    Parameters
+    ----------
+    dictionary : :class:`~gensim.corpora.dictionary.Dictionary`
+        A dictionary that specifies the considered terms.
+    alpha : float, optional
+        The multiplicative factor alpha defined by Charlet and Damnati (2017).
+    beta : float, optional
+        The exponential factor beta defined by Charlet and Damnati (2017).
+    threshold : float, optional
```

Comment:
Inconsistent naming with `max_distance` above.

Reply:
This should be resolved as per the above discussion.